data/src/main/java/pl/edu/agh/papaya/service/project/ProjectService.java

```diff
@@ -0,0 +1,64 @@
+package pl.edu.agh.papaya.service.project;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pl.edu.agh.papaya.model.Project;
+import pl.edu.agh.papaya.model.User;
+import pl.edu.agh.papaya.model.UserInProject;
+import pl.edu.agh.papaya.model.UserRole;
+import pl.edu.agh.papaya.repository.ProjectRepository;
+import pl.edu.agh.papaya.repository.UserInProjectRepository;
+import pl.edu.agh.papaya.security.UserContext;
+
+@Service
+public class ProjectService {
+
+    @Autowired
+    private ProjectRepository projectRepository;
+
+    @Autowired
+    private UserInProjectRepository userInProjectRepository;
+
+    @Autowired
+    private UserContext userContext;
+
+    public Optional<Project> getProjectById(Long id) {
+        return projectRepository.findById(id);
+    }
+
+    public List<Project> getUserProjects(String userId) {
+        long parsedUserId;
+        try {
+            parsedUserId = Long.parseLong(userId);
+        } catch (NumberFormatException e) {
+            return Collections.emptyList();
```

Comment:
A nie powinien tu właśnie polecieć jakiś bad_request?

Reply:
Good point, ale ogólnie średnio żeby stąd leciało bad request, bo to nie web, i raczej powinna lecieć jakaś 500 bo wtedy jest zalogowany user o złym ID. Tylko że wtedy znowu  przydałoby się sprawdzić czy user o ID numerycznym istnieje, bo wtedy mielibyśmy niekonsystentne zachowanie: user, który nie istnieje o ID numerycznym nie będzie mieć projektów, a o ID nienumerycznym zwróci 500. Więc w sumie nie wiem czy tego nie zostawić tak, ewentualnie uzupełnić dokumentację o opis tego zachowania. Co o tym myślisz?

Reply:
Dodałem więc zapiskę w dokumentacji.